core/src/test/scala/io/chrisdavenport/natchezlocal/TraceLocalSpec.scala

```diff
@@ -0,0 +1,33 @@
+package io.chrisdavenport.natchezlocal
+
+import cats.effect.{Trace => _, _}
+import io.chrisdavenport.fiberlocal.FiberLocal
+import munit.CatsEffectSuite
+import natchez.Span
+import natchez.noop.NoopSpan
+
+class TraceLocalSpec extends CatsEffectSuite {
+
+  test("TraceLocal.spanR should restore the FiberLocal span and not the span in place when the FunctionK is created") {
```

Comment:
@armanbilge if this test looks right to you, I think we could probably adapt it for use in Natchez itself, too

Reply:
Oh I guess there already is a test—I missed that when I was looking at this last night!

Reply:
I added a test in https://github.com/tpolecat/natchez/pull/681. Actually I was wondering if we should publish those as some form of laws.

Reply:
Yeah sorry I added it in a follow up 😅